OTAnalytics/plugin_datastore/track_store.py

```diff
@@ -670,18 +678,11 @@ def _get_dataset_with_classes(self, classes: list[str]) -> PandasTrackDataset:
         dataset = self._other.get_data()
         mask = dataset[track.TRACK_CLASSIFICATION].isin(classes)
         filtered_df = dataset[mask]
-        tracks_to_keep = filtered_df.index.get_level_values(LEVEL_TRACK_ID).unique()
-        tracks_to_remove = tracks_to_keep.symmetric_difference(
-            self._other.get_track_ids_as_string()
-        )
-        updated_geometry_datasets = self._other._remove_from_geometry_dataset(
-            tracks_to_remove
-        )
         return PandasTrackDataset(
-            self._other.track_geometry_factory,
-            filtered_df,
-            updated_geometry_datasets,
-            self._other.calculator,
+            track_geometry_factory=self._other.track_geometry_factory,
+            dataset=filtered_df,
+            geometry_datasets=None,
```

Comment:
I know we discussed that `geometry_datasets` should be `None` here. Please document why.

Reply:
I added a comment in the code. Please check if the explanation is acceptable.